File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
private:
    vector<vector<int>> ans;
    vector<int> current;
    void dfs(vector<int>& nums,int index){
        if(index == nums.size()){
            ans.push_back(current);
            return;
        }
        current.push_back(nums[index]);
        dfs(nums,index+1);

        current.pop_back();
        dfs(nums,index+1);

    }
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        ans.clear();
        current.clear();
        dfs(nums,0);
        return ans;
    }
};

File: Data Structures & Algorithms/sum-of-all-subset-xor-totals/submission-0.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        
        def dfs(index: int, current_xor: int) -> int:
            if index == len(nums):
                return current_xor
            
            withoutnum = dfs(index + 1,current_xor)
            withnum = dfs(index + 1,current_xor ^ nums[index])

            return withoutnum + withnum
        return dfs(0,0)